### ISSUE TITLE
Issue with kernel_gpu_opencl.cl#L120

### ISSUE THREAD

**Issue by streamhsa** [yuhc/gpu-rodinia#1]
It seems there is a bug in rodinia

https://github.com/yuhc/gpu-rodinia/blob/master/opencl/lavaMD/kernel/kernel_gpu_opencl.cl#L120

It uses variables in local address space in the non-outermost scope of a kernel function, which violates OpenCL spec v1.2 s6.5.2 Variables declared in the __local address space inside a kernel function must occur at kernel function scope. https://www.khronos.org/registry/OpenCL/specs/opencl-1.2.pdf 

 Can App Developer check and resolve this issue. Due to this, test is failing on OpenCL devices.

**Comment by yuhc** [comment]
Thanks for reporting this bug! It looks fine to move the definitions of those three local variables out of the if scope. I fixed it in commit bc053f8fd , and by the way fix a small mistake in opencl/b+tree Makefile.

**Comment by streamhsa** [comment]
Thanks. Does the link http://lava.cs.virginia.edu/Rodinia/download_links.htm has the updated compiled package rodinia_3.1.tar.bz2 with this fix so that I can take directly and test?
Or do we need to download the source and re-compile?

**Comment by yuhc** [comment]
Sorry, no... 😢  I'm not in that research lab and I'm not sure whether someone is still maintaining it. I've seen several papers using Rodinia, and based on their uses, here are two suggestions:
1. Download their package and maintain a local bug-free version. (That's also the reason why I created this repo)
2. Download this repo and copy `data/` from `rodinia_3.1.tar.bz2`. Then maintain a local bug-free version based on this version. I've already fixed several bugs in Rodinia in this repository, so many duplicated fixes can be avoided.
